src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -660,6 +665,9 @@ public void run() {
                             "Ignoring exception", ie);
                     }
                     closeSocket(client);
+                } catch (ConfigException ce) {
+                    LOG.error(ce.getMessage());
+                    throw new RuntimeException(ce);
```

Comment:
Don't we have to call `closeSocket(client)` here before throwing the `RuntimeException`? Relying on OS/JVM to close a socket can be too much optimistic and leak resources. If I am right (please double check) then line 667 can be moved to a finally block just below this catch clause.

Also, as the program flow is interrupted if a `ConfigException` is thrown lines 679-687 wouldn't be called and then `ServerSocket` also is not properly closed, right?

PS: I am not sure, would have to look carefully later and don't have time now, so excuse me if I am misunderstood it.